pkg/apis/extensions/validation/validation_test.go

```diff
@@ -1929,3 +1929,76 @@ func newBool(val bool) *bool {
 	*p = val
 	return p
 }
+
+func TestValidateStorageClass(t *testing.T) {
+	successCases := []extensions.StorageClass{
+		{
+			// Empty type and parameters
+			ObjectMeta:            api.ObjectMeta{Name: "foo"},
+			ProvisionerType:       "",
+			ProvisionerParameters: map[string]string{},
+		},
+		{
+			// nil parameters
+			ObjectMeta:      api.ObjectMeta{Name: "foo"},
+			ProvisionerType: "",
+		},
+		{
+			// some parameters
+			ObjectMeta:      api.ObjectMeta{Name: "foo"},
+			ProvisionerType: "kubernetes.io/foo-provisioner",
+			ProvisionerParameters: map[string]string{
+				"kubernetes.io/foo-parameter": "free/form/string",
+				"foo-parameter":               "free-form-string",
+				"foo-parameter2":              "{\"embedded\": \"json\", \"with\": {\"structures\":\"inside\"}}",
+			},
+		},
```

Comment:
It should be a success if there are valid params but no provisioner type according to how the code is written now -- is that desireable?